code/modules/modular_computers/file_system/programs/generic/wordprocessor.dm

```diff
@@ -108,6 +112,21 @@
 		var/newtext = sanitize(replacetext(input(usr, "Editing file '[open_file]'. You may use most tags used in paper formatting:", "Text Editor", oldtext) as message|null, "\n", "\[br\]"), MAX_TEXTFILE_LENGTH)
 		if(!newtext)
 			return
+
+		//Count the fields
+		var/laststart = 1
+		var/fields
+		while(TRUE)
```

Comment:
infinite loop with no sanity doesn't look that good
maybe use regex instead?
something like

```
	var/fields = 0
	var/regex/re = regex(@"\[field\]","g")
	while(re.Find(text))
		fields++
```